src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java

```diff
@@ -278,4 +278,52 @@ public Response bulkRemoveUsersFromDaa(
       return createExceptionResponse(e);
     }
   }
+
+  @POST
+  @Consumes(MediaType.APPLICATION_JSON)
+  @RolesAllowed({ADMIN, CHAIRPERSON})
+  @Path("/bulk/user/{userId}")
+  public Response bulkAddDAAsToUser(
+      @Auth AuthUser authUser,
+      @PathParam("userId") Integer userId,
+      String json) {
+    try {
+      User authedUser = userService.findUserByEmail(authUser.getEmail());
+      User user = userService.findUserById(userId);
+      if (authedUser.hasUserRole(UserRoles.SIGNINGOFFICIAL) && !Objects.equals(authedUser.getInstitutionId(), user.getInstitutionId())) {
+        return Response.status(Status.FORBIDDEN).build();
+      }
+      List<DataAccessAgreement> daaList = daaService.findDAAsInJsonArray(json, "daaList");
+      for (DataAccessAgreement daa : daaList) {
+        libraryCardService.addDaaToUserLibraryCardByInstitution(user, authedUser.getInstitutionId(), daa.getDaaId());
```

Comment:
Through local testing, I found that when trying to add a DAA that was already on the user we get a 409 in response, but no helpful logging. I think we should check if the user has any of the relationships already on their library card before making any more relationships so we don't end up in a halfway-state where some of the DAAs in the list get added and some don't.

Reply:
Let's discuss what the expected behavior should be.